catboost/libs/model/eval_processing.h

```diff
@@ -27,6 +27,19 @@ inline void CalcSoftmax(const TConstArrayRef<double> approx, TArrayRef<double> s
     }
 }
 
+inline TVector<double> CalcExponent(const TConstArrayRef<double> approx) {
+    TVector<double> exponents;
+    exponents.yresize(approx.size());
+
+    for (size_t dim = 0; dim < approx.size(); ++dim) {
+        exponents[dim] = approx[dim];
+    }
```

Comment:
It's better to take this `approx` argument by value. In this case you don't need to manual copy it:
```
inline TVector<double> CalcExponent(TVector<double> approx) {
    FastExpInplace(approx.data(), approx.ysize());
    return approx;
}
```